sanic/http/http3.py

```diff
@@ -333,7 +338,15 @@ def get_receiver_by_stream_id(self, stream_id: int) -> Receiver:
         return self.receivers[stream_id]
 
     def _make_request(self, event: HeadersReceived) -> Request:
-        headers = Header(((k.decode(), v.decode()) for k, v in event.headers))
+        try:
+            headers = Header(
+                (
+                    (k.decode("ASCII"), v.decode(errors="surrogateescape"))
+                    for k, v in event.headers
+                )
+            )
+        except UnicodeDecodeError:
+            BadRequest("Header names can only contain US-ASCII characters")
```

Comment:
Please revert this. It's done in the other PR #2710

Reply:
fixed